### ISSUE TITLE
[integration_test] Cannot run on iOS (please run "pub get" again)

### ISSUE THREAD

**Issue by creativecreatorormaybenot** [flutter/flutter#71515]
## Problem

I am really confused as to why the `flutter_driver` run `integration_test`s will not start on iOS.

You can find a **complete reproducible sample** (exact state of the Flutter project can be found in the [file tree of this commit](https://github.com/creativecreatorormaybenot/wakelock/tree/b62a023cd002640186686ad47f67dadedc029f02/wakelock/example)). This should run on CI, so you can even exactly _follow_ logs.

## Steps to Reproduce

You can check [this GitHub Actions run](https://github.com/creativecreatorormaybenot/wakelock/runs/1482974268) for the full steps.

1. Do the `integration_test` setup (already done in the [linked project](https://github.com/creativecreatorormaybenot/wakelock/tree/b62a023cd00264018668a6ad47f67dadedc029f02/wakelock/example)).
2. Run `flutter drive --driver=test_driver/integration_test.dart --target=integration_test/wakelock_test.dart --no-sound-null-safety`

**Expected results:** it should run the integration tests.

**Actual results:**

```
Running "flutter pub get" in example...                          1,169ms
Running pod install...                                              5.1s
Running Xcode build...                                          
Xcode build done.                                           132.9s
The pubspec.yaml file has changed since the pubspec.lock file was generated, please run "pub get" again.
Failed to stop app
```

>The pubspec.yaml file has changed since the pubspec.lock file was generated, please run "pub get" again.

### Full logs

You can find the full [raw logs here](https://pipelines.actions.githubusercontent.com/4qm0Lo3aw8465pHJ7vG2IfT2VyBpFp4Og4eNbiBrlCMacKVfp0/_apis/pipelines/1/runs/190/signedlogcontent/10?urlExpires=2020-12-01T22%3A48%3A11.6767855Z&urlSigningMethod=HMACV1&urlSignature=1VKv7VLJIRJz1iW8ehJrZmTliksuE4Ry29MM98TmYPQ%3D).  

They were run on https://github.com/creativecreatorormaybenot/wakelock/tree/b62a023cd00264018668a6ad47f67dadedc029f02/wakelock/example.

### Note

The same integration tests ran before, so I am not sure why the "please run 'pub get' again" message is popping up now ([example run](https://github.com/creativecreatorormaybenot/wakelock/runs/1304624169) that succeeded).  
I can also reproduce this on my *local machine*.

**Comment by jonahwilliams** [comment]
I think this should have been fixed with https://github.com/flutter/flutter/pull/70999 , but I can't tell what version you are on.

**Comment by creativecreatorormaybenot** [comment]
@jonahwilliams You can see it in the raw logs, but I should have included it:

It happens on both `1.24.0-10.2.pre-beta` and `1.24.0-10.2.pre-dev` in CI.
Locally, it also happens with `1.24.0-10.2.pre` `dev` channel.

→ the PR is not on these versions yet because `dev` and `beta` have not received updates in quite a while.

**Comment by creativecreatorormaybenot** [comment]
@jonahwilliams The Pub issue seems to be resolved on `master` 🎉  (see https://github.com/creativecreatorormaybenot/wakelock/pull/66/checks?check_run_id=1483575046).

I am now just having issues with null safety and Flutter driver, but that has nothing to do with this issue.  
→ you can close this now if you want. I will certainly do it once the fix lands on a channel.

**Comment by jonahwilliams** [comment]
Awesome!  🎊
